Add GH Action that builds PHAR files with build-release.sh

mlocati opened the pull request:
People may install pear in two ways (AFAIK):

1. By using the https://pear.php.net/install-pear-nozlib.phar file (for example [PHP uses it](https://github.com/php/php-src/blob/php-8.4.1/pear/Makefile.frag#L10C22-L11C1))
2. By using https://pear.php.net/go-pear.phar (as [described in the manual](https://pear.php.net/manual/en/installation.getting.php#installation.getting.unix))

What about adding a GitHub Action that generates those two PHAR files (and attaches them to GitHub Releases autimatically when someone publishes a [new release](https://github.com/pear/pear-core/releases)).

We'd also need to update the two PHARs on the pear.php.net website, but at least we don't have to compile them manually.

CloCkWeRX reviewed (approved):
Conceptually; great.

Probably warrants a little bit of security review around `shivammathur/setup-php@v2` or any of the other actions; which we do already use; but _not_ to create an artifact potentially many folks will access that persists beyond the life of a CI job.

May need to look at more exact version pinning/sha1 or something like that (https://github.com/shivammathur/setup-php/releases/tag/v2 for example says it was 'updated')

mlocati commented:
> Probably warrants a little bit of security review around `shivammathur/setup-php@v2`

I've updated this PR with 060f823e964493844e8919e6b02750f0b0b63ad7

> or any of the other actions

The other actions are `actions/checkout` and `actions/upload-artifact` which are maintained by GitHub: should we use specific commits for them too?

CloCkWeRX commented:
IMO yes, when I think about stuff like https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash
Dependabot... Should update it I think; so it's not a huge maintenance overhead, though may be a bit annoying as we are fairly low traffic vs volume of PRs. 

Will let a few of the other maintainers add their thoughts

mlocati commented:
> IMO yes

Done - I've added c49eb6948c9d2bd45b051ff7958fd5d348e66bd9

I've also added 8bef117ec87435ef4abbe0e1b354e9139c261cd0 and c72b14b9b26ae028e5374d2324b431639257955f

mlocati commented:
We could also improve the way we publish GH releases...
We could switch to something like this:
1. maintainers push a version-like tag (that is, something that matches `/^(v\.?)?\d+.\d+.\d+/`)
2. a GH action detects that we have a new release to be published
3. it builds the assets
4. il publish a new GH release

I'm already doing this for https://github.com/mlocati/powershell-phpmanager (and somehow also for https://github.com/mlocati/docker-php-extension-installer)

ashnazg commented:
Thanks for the effort on this, but for now, I'd prefer to continue generating _and signing_ them myself.  Your PR will be a useful reference to later implement this though, so it is definitely appreciated (and might get reopened/merged later).

mlocati commented:
@ashnazg what about keeping the GH workflow that builds the PHARs? That way we check if everything works as expected (and people have a working reference to see how to build them)
